From 51207ace580a47ef45bf7d459810978283ac3b62 Mon Sep 17 00:00:00 2001
From: Tomas Zak <tomas.zak@turris.com>
Date: Thu, 25 Jan 2024 11:32:22 +0100
Subject: [PATCH] packages: openvpn: hotfix for openvpn client's side

Hotfix openvpn for client's side, wrong path of configuration file were
provided.
This fixing commit: 4f7e6ba76544895a63bd7b9074ecf6fad8b16d74
---
 net/openvpn/files/openvpn.init | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/openvpn/files/openvpn.init b/net/openvpn/files/openvpn.init
index de0f85aa0..580b0d36b 100644
--- a/net/openvpn/files/openvpn.init
+++ b/net/openvpn/files/openvpn.init
@@ -140,14 +140,14 @@ openvpn_get_credentials() {
 openvpn_add_instance() {
 	local name="$1"
 	local dir="$2"
-	local conf=$(basename "$3")
+	local conf="$3"
 	local security="$4"
 	local up="$5"
 	local down="$6"
 	local route_up="$7"
 	local route_pre_down="$8"
 	local ipchange="$9"
-	local client=$(grep -qEx "client|tls-client" "$dir/$conf" && echo 1)
+	local client=$(cd "$dir"; grep -qEx "client|tls-client" "$conf" && echo 1)
 
 	procd_open_instance "$name"
 	procd_set_param command "$PROG"	\
-- 
2.44.0

